src/graph/backend/dnnl/thread_local_cache.hpp

```diff
@@ -181,14 +181,16 @@ class thread_local_cache_t {
         static global_cache_type_t *get_global_cache() {
             // A global table to store cached values in ALL threads. This global
             // table takes the ownership of cached values
-            try {
-                static auto global_cache = std::shared_ptr<global_cache_type_t>(
-                        new global_cache_type_t {},
-                        [](global_cache_type_t *ptr) {
-                            return ptr->release();
-                        });
-                return global_cache.get();
-            } catch (...) { return nullptr; }
+            static std::once_flag flag;
+            static global_cache_type_t *global_cache = nullptr;
+
+            std::call_once(flag, []() {
+                try {
```

Comment:
Given the shared_ptr is removed, i think we don't need the try-catch anymore.

Reply:
the `new` operation may also throw, e.g. bad_alloc, in that case try/cache can help